engine/v2/data/interface.go

```diff
@@ -0,0 +1,13 @@
+package data
+
+import (
+	"context"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/data/datamodels"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+)
+
+// Repo is the interface that wraps the basic operations with data.
+type Repo interface {
+	FindForUser(ctx context.Context, outpoint bsv.Outpoint, userID string) (*datamodels.Data, error)
```

Comment:
Maybe it should be ID instead of outpoint, and should be simply a string?
WDYT?
Because now it's that low level way of how we build an ID is exposed to all layers